Resources/public/js/views/ez-dashboardview.js

```diff
@@ -33,35 +39,161 @@ YUI.add('ez-dashboardview', function (Y) {
+                if (value[key] instanceof Y.eZ.DashboardBlockView) {
+                    blocks[key] = value[key];
+                }
+            });
+
+            return blocks;
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param identifier {String} block identifier
+         * @param view {Y.eZ.DashboardBlockView} dashboard block view instance
+         * @return {Object} the hash of blocks
+         */
+        addBlock: function (identifier, view) {
+            var blocks = this.get('blocks');
+
+            if (blocks[identifier]) {
+                blocks = this.removeBlock(identifier, true);
+            }
+
+            blocks[identifier] = view;
 
+            this.set('blocks', blocks);
+
+            return blocks;
+        },
+
+        /**
+         * Removes a block from the dashboard
+         *
+         * @method removeBlock
+         * @param identifier {String} block identifier
+         * @param preventUpdating {Boolean} should the `blocks` attribute value be updated?
+         * @return {Object} the hash of blocks
+         */
+        removeBlock: function (identifier, preventUpdating) {
+            var blocks = this.get('blocks'),
+                block = blocks[identifier];
+
+            block.removeTarget(this);
+            block.destroy({remove: true});
+
+            delete blocks[identifier];
+
+            if (!preventUpdating) {
+                this.set('blocks', blocks);
+            }
+
+            return blocks;
+        },
+
+        /**
+         * Hides a block on the dashboard
+         *
+         * @method hideBlock
+         * @param identifier {String} block identifier
+         * @return {eZ.DashboardView} the view itself
+         */
+        hideBlock: function (identifier) {
+            this.get('blocks')[identifier].set('displayed', false);
+
+            return this;
+        },
+
+        /**
+         * Shows a block on the dashboard
+         *
+         * @method showBlock
+         * @param identifier {String} block identifier
+         * @return {eZ.DashboardView} the view itself
+         */
+        showBlock: function (identifier) {
+            this.get('blocks')[identifier].set('displayed', true);
+
+            return this;
+        },
     }, {
         ATTRS: {
             /**
-             * Stores the iframe Source
+             * Hash containing information about all blocks in the dashboard
              *
-             * @attribute iframeSource
-             * @type String
-             * @default 'http://ez.no/in-product/eZ-Platform'
-             * @readOnly
+             * @attribute blocks
+             * @type Object
+             * @example {
+             *     key1<String>: viewInstance1<Y.eZ.DashboardBlockView>,
+             *     key2<String>: viewInstance2<Y.eZ.DashboardBlockView>,
+             * }
              */
-            iframeSource: {
```

Comment:
for the record, you can not remove an attribute like that, that's a BC break. That's why to easily avoid that, you should create another view implementation for the dashboard.